floss/features/extract.py

```diff
@@ -116,32 +117,75 @@ def extract_function_kinda_tight_loop(f):
     Yields tight loop features in the provided function
     Algorithm by Blaine S.
     """
+    cfg = viv_utils.CFG(f)
+
+    root_bb_va = cfg.get_root_basic_block()
+    leaf_bb_vas = {bb.va for bb in cfg.get_leaf_basic_blocks()}
+
     for bb in f.basic_blocks:
         # skip first and last BBs
-        if bb.va == f.va or bb.va + bb.size == f.basic_blocks[-1].va:
```

Comment:
bug here that the "last" basic block may not be end of the function. 

now we use the "leaves" that should be the ends of the function.